src/pages/questions/ProductInformation.jsx

```diff
@@ -42,6 +43,21 @@ const ProductInformation = ({ question }) => {
   const { t } = useTranslation();
   const [productData, setProductData] = React.useState({});
   const [hideImages, setHideImages] = React.useState(getHideImages);
+  const [flagged, setFlagged] = React.useState([]);
+
+  const flagImage = (src, barcode, index) => {
+    const response = updateFlagImage(barcode, src);
```

Comment:
considering that the `updateFlagImage` is doing an asyc call, you may want to do the `flagImage` an `async` ( ...`flagImage = async (...) => { ... }` ) function, and `await` the response of the `updateFlagImage`

Reply:
Since that might take some time, I thought it would be better to work in the background to improve the user experience.

Reply:
ohh I see, that's a nice idea, something on the lines of [optimistic updates](https://stackoverflow.com/questions/33009657/what-is-optimistic-updates-in-front-end-development), I think as you said it is good UX. An option would be to keep it as you have it, but please add the `updateFlagImage(...).catch()` and in the `catch` you would need to set it back to the original state. Another option would be to wait and show a loading spinner in the button, I think in most cases it shouldn't take too long.